radixdlt/src/main/java/com/radixdlt/consensus/sync/BFTSync.java

```diff
@@ -248,25 +248,35 @@ private void doCommittedSync(SyncState syncState) {
 
 	@Override
 	public void processGetVerticesLocalTimeout(LocalGetVerticesRequest request) {
-		SyncRequestState syncRequestState = bftSyncing.get(request);
+		SyncRequestState syncRequestState = bftSyncing.remove(request);
 		if (syncRequestState == null) {
 			return;
 		}
-		int nextIndex = random.nextInt(syncRequestState.authors.size());
-		BFTNode nextNode = syncRequestState.authors.get(nextIndex);
-		this.requestSender.sendGetVerticesRequest(nextNode, request);
-		this.timeoutScheduler.scheduleTimeout(request, bftSyncPatienceMillis);
+
+		List<HashCode> syncIds = syncRequestState.syncIds.stream().filter(syncing::containsKey).collect(Collectors.toList());
+		for (HashCode syncId : syncIds) {
+			SyncState syncState = syncing.get(syncId);
+			if (syncState == null) {
+				return;
+			}
+
+			// Retry full sync on timeout
+			int nextIndex = random.nextInt(syncRequestState.authors.size());
```

Comment:
I think we have a problem here.  I can't find a binding for Random.class, which means that this is going to be bound to a default constructed version of java.util.Random.

We should be using a binding to SecureRandom somewhere.

Reply:
yup good point